### PR TITLE
The err in the else branch can not be returned in the end

### PR DESCRIPTION
**Pull request by YuPengZTE**
Signed-off-by: yupengzte <yu.peng36@zte.com.cn>

### CODE REVIEW

**Comment by allencloud** [comment]
Could you explain more? @YuPengZTE

**Comment by YuPengZTE** [comment]
@allencloud 
Before change, 
[cluster\swarm\cluster.go#L579](https://github.com/docker/swarm/blob/master/cluster/swarm/cluster.go#L579): ineffectual assignment to err,
[cluster\swarm\cluster.go#L582](https://github.com/docker/swarm/blob/master/cluster/swarm/cluster.go#L582): ineffectual assignment to err,
because, in [cluster/swarm/cluster.go#L571](https://github.com/docker/swarm/blob/master/cluster/swarm/cluster.go#L571)，the ***err*** overload the ***err*** of [cluster/swarm/cluster.go#L540](https://github.com/docker/swarm/blob/master/cluster/swarm/cluster.go#L540)

File: cluster/swarm/cluster.go [hunk 1]
```diff
@@ -568,9 +568,9 @@ func (c *Cluster) CreateVolume(request *volume.VolumesCreateBody) (*types.Volume
 		wg.Wait()
 	} else {
 		config := cluster.BuildContainerConfig(containertypes.Config{Env: []string{"constraint:node==" + parts[0]}}, containertypes.HostConfig{}, networktypes.NetworkingConfig{})
-		nodes, err := c.scheduler.SelectNodesForContainer(c.listNodes(), config)
-		if err != nil {
-			return nil, err
+		nodes, errs := c.scheduler.SelectNodesForContainer(c.listNodes(), config)
+		if errs != nil {
+			return nil, errs
```
Comment (dongluochen) [inline cluster/swarm/cluster.go]: @YuPengZTE Good catch! I don't want to introduce yet another variable to deal with this. How about removing `:=` like this? 
```
                var nodes []*node.Node
		nodes, err = c.scheduler.SelectNodesForContainer(c.listNodes(), config)
 		if err != nil {
 			return nil, err
```

Comment (YuPengZTE) [inline cluster/swarm/cluster.go]: Good idea!
Changes had done, PTAL. 
@dongluochen , @allencloud .

**Comment by dongluochen** [comment]
LGTM

**Comment by allencloud** [comment]
Thanks for your contribution. @YuPengZTE  ☀️

**Pull request merged**